virttest/utils_test/libvirt.py

```diff
@@ -2092,6 +2092,51 @@ def create_nwfilter_xml(params):
         raise exceptions.TestFail("Fail to create nwfilter XML: %s" % detail)
 
 
+def get_channel_info(session, interface):
+    """
+    Get channel parameters of interface in vm.
+    """
+    cmd = "ethtool -l %s" % interface
+    s, o = session.cmd_status_output(cmd)
+    if s:
+        logging.info("Get channel parameters for vm failed:%s" % o)
+        return {}, {}
+    maximum = {}
+    current = {}
+    # Just for temp
+    settings = {}
+    for line in o.splitlines():
+        if line.count("maximums"):
+            settings = maximum
+        elif line.count("Current"):
+            settings = current
+        parameter = line.split(':')[0].strip()
+        if parameter in ["RX", "TX", "Other", "Combined"]:
+            settings[parameter] = line.split(':')[1].strip()
+    return maximum, current
+
+
+def setting_channel(session, interface, parameter, value):
+    """
+    Setting channel parameters for interface.
+    """
+    cmd = "ethtool -L %s %s %s" % (interface, parameter, value)
+    logging.debug("The cmd to set the channel info %s" % cmd)
+    s, o = session.cmd_status_output(cmd)
+    if s:
+        logging.debug("Setting %s to %s failed:%s", parameter, value, o)
+        return False
+    maximum, current = get_channel_info(session, interface)
+    logging.debug("After set, the current value is %s" % current)
+    try:
+        if int(current['Combined']) == int(value):
```

Comment:
'Combined' should be 'parameter' instead?

Reply:
no, 'Combined' is the key of the dict "current".